src/preact/base-element.js

```diff
@@ -499,8 +638,10 @@ export class PreactBaseElement extends BaseElement {
   onReadyState_(state, opt_failure) {
     this.setReadyState?.(state, opt_failure);
 
-    const Ctor = this.constructor;
-    if (Ctor['unloadOnPause']) {
+    const Ctor = /** @type {typeof PreactBaseElement} */ (
+      /** @type {unknown} */ (this.constructor)
```

Comment:
```suggestion
      /** @type {?} */ (this.constructor)
```

Reply:
I've seen this a ton. Why is the returntype of `this.constructor` just `Element` instead of `typeof PreactBaseElement`?

Reply:
relevant: https://github.com/microsoft/TypeScript/issues/3841.
idk

Reply:
What I meant to suggest by this was, can we change `constructor` to explicitly return `typeof PreactBaseElement` so we don't need this cast everywhere it gets used?

Reply:
We could always create a coercion function if you don't like the noise? Imo the current workaround is fine, and in TS will stay to a single line anway.

Reply:
le sigh, no, no coercion function necessary. I was just hoping there was some way to teach TS to handle it correctly, but then you reminded me TS is just ugly sometimes 😆